inc/notificationtarget.class.php

```diff
@@ -677,32 +677,69 @@ public function addItemAuthor() {
 
       // members/managers of the group allowed on object entity
       // filter group with 'is_assign' (attribute can be unset after notification)
-      $query = $this->getDistinctUserSql()."
-               FROM `glpi_groups_users`
-               INNER JOIN `glpi_users` ON (`glpi_groups_users`.`users_id` = `glpi_users`.`id`) ".
-               $this->getProfileJoinSql()."
-               INNER JOIN `glpi_groups` ON (`glpi_groups_users`.`groups_id` = `glpi_groups`.`id`)
-               WHERE `glpi_groups_users`.`groups_id` = '$group_id'
-                     AND `glpi_groups`.`is_notify`";
+      $criteria = array_merge(
+         [
+            'INNER JOIN' => [
+               User::getTable() => [
+                  'ON' => [
+                     Group_User::getTable()  => 'users_id',
+                     User::getTable()        => 'id'
+                  ]
+               ]
+            ]
+         ],
+         $this->getDistinctUserCriteria() + $this->getProfileJoinCriteria()
+      );
+      $criteria['FROM'] = Group_User::getTable();
+      $criteria['INNER JOIN'][Group::getTable()] = [
```

Comment:
Why this one is disctint from the above array_merge ?

Reply:
The 2 were previously there... And I've moved only one :p